__main__.py

```diff
@@ -35,5 +38,16 @@ def build() -> None:
     runpy.run_module('coherent.build', run_name='__main__')
 
 
+@app.command(context_settings=dict(allow_extra_args=True))
+def tag(
+    tag_name: str,
+    context: typer.Context,
+    location: Annotated[str, typer.Option('-C', help='Path to repository.')] = '.',
```

Comment:
I'm slightly disinclined to build on the git convention of `-C`. It's not a very good indicator of the "repository". Maybe the default "--location" would be sufficient. Or maybe we should follow the [jj convention of -R/--repository](https://martinvonz.github.io/jj/latest/cli-reference/#jj).

The more I think about it, I like "repository" as the parameter... and you can make the type a `jaraco.vcs.repo` instead of a `str`, and then in the git command, pass `repository.location`.

Eventually, I'd like to support more than just git repos and I plan to do that through `jaraco.vcs` abstractions, but currently the Repo object doesn't have a method to create tags.

Reply:
`jaraco.vcs.Repo`*
I think it would need to be annotated as `Annotated[Repo, BeforeValidator(Repo.detect)]` (hoping that `Repo.detect()` is redundant when passed a `Repo` object).